python/cugraph/cugraph/tests/test_sorensen.py

```diff
@@ -0,0 +1,239 @@
+
+import gc
+import time
+import pytest
+
+import cudf
+import cugraph
+from cugraph.tests import utils
+from pathlib import PurePath
+
+# Temporarily suppress warnings till networkX fixes deprecation warnings
+# (Using or importing the ABCs from 'collections' instead of from
+# 'collections.abc' is deprecated, and in 3.8 it will stop working) for
+# python 3.7.  Also, this import networkx needs to be relocated in the
+# third-party group once this gets fixed.
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import networkx as nx
+
+
+print("Networkx version : {} ".format(nx.__version__))
+
+
+def cugraph_call(cu_M, edgevals=False):
+    G = cugraph.Graph()
+    if edgevals is True:
+        G.from_cudf_edgelist(cu_M, source="0", destination="1", edge_attr="2")
+    else:
+        G.from_cudf_edgelist(cu_M, source="0", destination="1")
+
+    # cugraph sorensen Call
+    t1 = time.time()
+    df = cugraph.sorensen(G)
+    t2 = time.time() - t1
+    print("Time : " + str(t2))
+
+    df = df.sort_values(["source", "destination"]).reset_index(drop=True)
+
+    return (
+        df["source"].to_array(),
+        df["destination"].to_array(),
+        df["sorensen_coeff"].to_array(),
+    )
+
+
+def networkx_call(M):
+
+    sources = M["0"]
+    destinations = M["1"]
+    edges = []
+    for i in range(len(M)):
+        edges.append((sources[i], destinations[i]))
+        edges.append((destinations[i], sources[i]))
+    edges = list(dict.fromkeys(edges))
+    edges = sorted(edges)
+    # in NVGRAPH tests we read as CSR and feed as CSC, so here we doing this
+    # explicitly
+    print("Format conversion ... ")
+
+    Gnx = nx.from_pandas_edgelist(
+        M, source="0", target="1", edge_attr="weight", create_using=nx.Graph()
+    )
+
+    # Networkx Jaccard Call
+    print("Solving... ")
+    t1 = time.time()
+    preds = nx.jaccard_coefficient(Gnx, edges)
+    t2 = time.time() - t1
+
+    print("Time : " + str(t2))
+    src = []
+    dst = []
+    coeff = []
+    for u, v, p in preds:
+        src.append(u)
+        dst.append(v)
+        # Conversion from Networkx Jaccard to Sorensen
+        # No networkX equivalent
+        coeff.append((2*p)/(1+p))
+    return src, dst, coeff
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_UNDIRECTED)
+def test_sorensen(graph_file):
+    gc.collect()
+
+    M = utils.read_csv_for_nx(graph_file)
+    cu_M = utils.read_csv_file(graph_file)
+    cu_src, cu_dst, cu_coeff = cugraph_call(cu_M)
+    nx_src, nx_dst, nx_coeff = networkx_call(M)
+
+    # Calculating mismatch
+    err = 0
+    tol = 1.0e-06
+
+    assert len(cu_coeff) == len(nx_coeff)
+    for i in range(len(cu_coeff)):
+        if abs(cu_coeff[i] - nx_coeff[i]) > tol * 1.1:
+            err += 1
+
+    print("Mismatches:  %d" % err)
+    assert err == 0
+
+
+@pytest.mark.parametrize("graph_file", [PurePath(
+    utils.RAPIDS_DATASET_ROOT_DIR)/"netscience.csv"]
+)
```

Comment:
if you use `utils.RAPIDS_DATASET_ROOT_DIR_PATH`, you don't need to explicitly create a `Path` object since it already is one.
```suggestion
@pytest.mark.parametrize("graph_file",
                         [utils.RAPIDS_DATASET_ROOT_DIR_PATH/"netscience.csv"]
                         )
```

Reply:
Fixed but I had to cast `"netscience.csv"` to `PurePath `though if not there was an error

Reply:
your recent changes didn't use `RAPIDS_DATASET_ROOT_DIR_PATH` as was done in the diff above.